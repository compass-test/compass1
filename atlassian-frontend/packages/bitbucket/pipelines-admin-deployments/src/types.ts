import { Environment } from '@atlassian/pipelines-models';

export type ReorderEnvironments = (
  type: string,
  environmentTypeOrder: { uuid: string }[],
) => void;

export type FetchBranches = (
  name: string,
) => Promise<{ values: { name: string }[] }>;

export type CreateBranchRestriction = (
  environmentUuid: string,
  pattern: string,
) => void;

export type DeleteBranchRestriction = (
  environmentUuid: string,
  restrictionUuid: string,
) => void;

export type CreateEnvironment = (
  environmentName: string,
  environmentType: string,
) => void;

export type UpdateEnvironment = (environmentUuid: string, change: any) => void;

export type DeleteEnvironment = (environmentUuid: string) => void;

export type RenderVariableList = (environment: Environment) => React.ReactNode;
