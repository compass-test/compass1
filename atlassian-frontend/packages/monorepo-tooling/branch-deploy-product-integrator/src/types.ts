export type Command = 'push' | 'clone-product-repo';
export type ErrorType = 'cli' | Command;
export interface ValidationError extends Error {
  type: ErrorType;
}

export type BranchMetadata = {
  /** The atlassian-frontend commit hash that was installed into the product branch */
  afCommitHash: string;
  /** The atlassian-frontend branch name */
  afBranchName: string;
  /** Whether the branch has been manually modified (committed to) or not */
  branchIsModified?: boolean;
};
