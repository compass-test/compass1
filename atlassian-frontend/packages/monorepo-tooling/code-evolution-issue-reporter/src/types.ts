export interface BaseRule {
  ruleName: string;
  type: 'eslint' | 'regex';
  title: string;
  description: string;
  helpLink: string;
}

export interface RegexRule extends BaseRule {
  fileGlob: string;
  match: string[];
}

export interface EslintRule extends BaseRule {
  reportedMessage?: string;
}

export interface EslintRuleMessage {
  ruleId: string;
  severity: number;
  message: string;
  line: number;
  column: number;
  nodeType: string;
  messageId: string;
  endLine: number;
  endColumn: number;
}

interface EslintDeprecatedRule {
  ruleId: string;
  replacedBy: string[];
}

export interface EslintFileReport {
  filePath: string;
  messages: EslintRuleMessage[];
  errorCount: number;
  warningCount: number;
  fixableErrorCount: number;
  fixableWarningCount: number;
  source: string;
  usedDeprecatedRules: EslintDeprecatedRule[];
}

export type RuleType = EslintRule | RegexRule;

type LineRange = {
  from: number;
  to: number;
};

export type FileResult = {
  name: string;
  lines: LineRange[];
};

export interface RuleResult {
  scope: string;
  assignee: string;
  team: string;
  fileResults: FileResult[];
}

export interface ReporterResultWithMeta {
  rule: RuleType;
  result: RuleResult[];
}

export interface ScannerResult {
  rule: RuleType;
  files: FileResult[];
}

export function isRegexRule(rule: EslintRule | RegexRule): rule is RegexRule {
  return rule.type === 'regex';
}

export function isEslintRule(rule: EslintRule | RegexRule): rule is EslintRule {
  return rule.type === 'eslint';
}

export type CliFlags = {
  eslintOutputFile?: string;
  configPath?: string;
};
