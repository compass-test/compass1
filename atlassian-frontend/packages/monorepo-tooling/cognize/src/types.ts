import { types as t, NodePath } from '@babel/core';

export type ArgsvParse = {
  [x: string]: unknown;
  c?: string | undefined;
  l?: string | undefined;
  d?: string | undefined;
  t?: string | undefined;
  i?: string | undefined;
  _?: string[];
  $0?: string;
};

type PluginOptions = {
  targetImports: string;
  customLogger?: Function | string | null;
  trackOnClient?: boolean;
  customVisitors?: CognizePlugin | string;
  ignoreDirs?: string;
};

export type BabelPluginsArray = BabelPlugin[];

type CognizePlugin = (args: BabelVisitor) => BabelVisitor;

export type BabelPluginState = {
  opts: PluginOptions & {
    filename: string;
    root: string;
    plugins: BabelPluginsArray;
    sourceFileName: string;
  };
  file: {
    code: string;
  };
  filename: string;
  fileNameIdentifier: t.Identifier;
  code: string;
};

export type PluginCache = Map<string, any>;

export type BabelVisitor = {
  file?: BabelPluginState;
  dataCache?: PluginCache;
  visitorCache?: PluginCache;
  targetedImportsMap?: PluginCache;
  Program?: {
    enter?: (arg0: any, arg1: BabelPluginState) => void;
    exit?: (arg0: any, arg1: BabelPluginState) => void;
    visitorCache?: PluginCache;
  };
  ImportDeclaration?: (arg0: Path, arg1: BabelPluginState) => void;
  CallExpression?: (arg0: Path, arg1: BabelPluginState) => void;
  MemberExpression?: (arg0: Path, arg1: BabelPluginState) => void;
  JSXElement?: (arg0: Path, arg1: BabelPluginState) => void;
  JSXOpeningElement?: (arg0: Path, arg1: BabelPluginState) => void;
  ExportDefaultDeclaration?: (arg0: Path, arg1: BabelPluginState) => void;
  ExportNamedDeclaration?: (arg0: Path, arg1: BabelPluginState) => void;
  ReturnStatement?: (arg0: Path, arg1: BabelPluginState) => void;
};

export type BabelPlugin = {
  key?: string;
  name: string;
  file?: BabelPluginState;
  dataCache?: PluginCache;
  visitorCache?: PluginCache;
  targetedImportsMap?: PluginCache;
  manipulateOptions?: (arg0: any, arg1: any) => void | undefined;
  inherits?: (arg0: string) => void;
  pre?: (arg0: BabelPluginState) => void;
  post?: (arg0: BabelPluginState) => void;
  visitor?: BabelVisitor;
  options?: PluginOptions;
  opts?: PluginOptions;
};

export type Path = NodePath & {
  parentPath: Path;
  hub: { file: BabelPluginState; getScope: () => Path['scope'] };
  scope: {
    generateUidIdentifier: (str: string) => void;
  };
  container: { openingElement: t.JSXOpeningElement };
  node: {
    type: string;
    argument: Path['node'];
    source: {
      value: string;
    };
    callee: {
      name: string;
    };
    openingElement: {
      name: {
        name: string;
      };
      attributes: t.JSXAttribute[];
    };
    loc: {
      start: {
        line: number;
      };
    };
    name: { name: string };
    object: { name: string };
    property: string;
    start: number;
    end: number;
    specifiers: [];
    value: string;
  };
  get: (arg0: string) => Path;
  stop: () => void;
  unshiftContainer: (str: string, declaration: any) => void;
  pushContainer: (str: string, declaration: any) => void;
  replaceWith: (ast: any) => void;
  traverse: (visitor: BabelVisitor) => void;
  findParent: (parentPath: object) => object;
};

export type ObjectLiteral = { [key: string]: any };
