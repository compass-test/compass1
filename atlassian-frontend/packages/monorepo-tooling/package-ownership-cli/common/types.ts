import {
  AFPackage as PackageInfo,
  TeamsJson,
} from '@atlaskit/build-utils/types';

export type { PackageInfo, TeamsJson };

type TeamInfo = Partial<TeamsJson['teamName']> & {
  packages: string[];
};

/** Extends teams.json with list of packages */
export type TeamMap = {
  [teamName: string]: TeamInfo;
};
