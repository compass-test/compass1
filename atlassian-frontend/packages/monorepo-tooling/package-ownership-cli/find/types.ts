export type { PackageInfo, TeamMap, TeamsJson } from '../common/types';

export const NO_TEAM = 'no team';

export type Options = {
  /** String of comma separated package names */
  packages?: string;

  /** Name of a team, as can be found in teams.json */
  team?: string;

  /** Include packages that include this substring, or whose paths match this glob */
  include?: string;

  /** Exclude packages that include this substring, or whose paths match this glob */
  exclude?: string;
};

export type CliFlags = Options & {
  /** Log what is happening to stdout and stderr */
  verbose: boolean;

  /** Format for CLI output, 'json' | 'text' | 'adf' */
  format: string;

  /** Path to file */
  output?: string;
};
