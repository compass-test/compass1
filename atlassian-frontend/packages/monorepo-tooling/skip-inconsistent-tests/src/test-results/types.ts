import { Test } from '../types';

export type TestResults = {
  // Total physical files run
  totalFiles: number;
  // Total test cases run (can be several per file)
  totalTestCases: number;
  // Total test case failures (included exempt ones)
  totalFailedTestCases: number;
  // The count non-exempt files that failed
  numSkippableFiles: number;
  // The count of non-exempt test cases that failed
  numSkippableTestCases: number;
  // The count of exempt test cases that failed but weren't skipped
  numExemptTestCases: number;
  // Non exempt failures ready for skipping
  skippableTestCases: Test[];
  // Exempt errors that shouldn't be skipped
  exemptTestCases: Test[];
};
