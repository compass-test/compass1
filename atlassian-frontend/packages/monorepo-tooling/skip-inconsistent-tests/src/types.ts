export type Test = {
  path: string;
  testName: string;
  ancestorLabels: string;
  pipelineStepId?: string;
  pullRequestUrl?: string;
  errors: string[];
  skipped?: boolean;
};

export type AggregatedSkippedTests = {
  vr?: Test[];
  integration?: Test[];
  mobile?: Test[];
  pullRequestUrl?: string;
  total: number;
};

// Multi test types per package. Final step.
export type PackageAggregatedSkippedTests = {
  [packageName: string]: AggregatedSkippedTests;
};

// Single test type (e.g. VR only, or integration only) per package. Intermediary step.
export type PackageSkippedTests = {
  [packageName: string]: Test[];
};

export type TestType = 'vr' | 'integration' | 'mobile';

export type JiraTicket = {
  key: string;
  browseUrl: string;
  assignee: string;
  attachment?: string;
};

export type Verbosity = 'none' | 'low' | 'high';
