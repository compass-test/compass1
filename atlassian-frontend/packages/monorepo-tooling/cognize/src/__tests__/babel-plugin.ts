import pluginTester, { TestObject } from 'babel-plugin-tester';

import path from 'path';

import babelPlugin from '../babel-plugin';

import { PluginCache, BabelPlugin } from '../types';

const commonOpts = {
  plugin: babelPlugin,
  pluginName: '@atlaskit/cognize',
  babelOptions: require('../test-utils/babel.config.js'),
  formatResult: (s: any) => s,
  snapshot: true,
};

const logger = (cache: PluginCache) => Array.from(cache.entries());

let customLogger: jest.Mock = jest.fn();
let consoleLogSpy: jest.SpyInstance = jest.fn();

describe('Plugin options', () => {
  beforeEach(() => {
    customLogger = customLogger.mockImplementation(logger);
    consoleLogSpy = jest.spyOn(console, 'log');
  });

  afterEach(() => {
    customLogger.mockReset();
  });

  pluginTester({
    ...commonOpts,
    tests: [
      {
        title: 'does not call custom logger if not passed through options',
        code: `import React from 'react';
import { B200, N0, N70A } from '@atlaskit/theme/colors';
export default {
  B200,
  N0,
  N70A
};`,
        pluginOptions: {
          targetImports: '@atlaskit/theme/colors',
        },
        setup() {
          consoleLogSpy.mockImplementationOnce(() => {});
        },
        teardown() {
          expect(consoleLogSpy).toBeCalled();
          expect(customLogger).toBeCalledTimes(0);
        },
      } as TestObject,
      {
        title: 'loads custom visitor',
        code: `import React from 'react';
import { B200, N0, N70A } from '@atlaskit/theme/colors';
const myFunc = () => B200();
myFunc();
export default {
  B200,
  N0,
  N70A
};`,
        pluginOptions: {
          targetImports: '@atlaskit/theme/colors',
          customVisitors: path.join(__dirname, '../test-utils/my-visitors.js'),
          customLogger,
        },
        teardown() {
          expect(customLogger).toBeCalledWith(
            new Map().set('', {
              BaseProjectName: '@atlaskit/fabric',
              DependencyVersions: {},
              File: '',
              CallExpressions: [
                {
                  end: 109,
                  extra: undefined,
                  innerComments: null,
                  leadingComments: null,
                  loc: {
                    end: {
                      column: 25,
                      line: 3,
                    },
                    filename: undefined,
                    identifierName: 'B200',
                    start: {
                      column: 21,
                      line: 3,
                    },
                  },
                  name: 'B200',
                  range: undefined,
                  start: 105,
                  trailingComments: null,
                  type: 'Identifier',
                },
              ],
              ImportDeclarations: {
                '@atlaskit/theme/colors': ['B200', 'N0', 'N70A'],
              },
              Project: '@atlaskit/fabric',
            }),
          );
        },
      } as TestObject,
    ],
  });
  pluginTester({
    ...commonOpts,
    filename: __filename,
    tests: [
      {
        title: 'does not parse file if `ignoreDirs` matches filename',
        fixture: '../__fixtures__/__ignored__/ignored.js',
        pluginOptions: {
          targetImports: '@atlaskit/theme/colors',
          ignoreDirs: '(__ignored__)',
          customLogger,
        },
        teardown() {
          expect(customLogger).toBeCalledTimes(0);
        },
      } as TestObject,
    ],
  });
});

describe('Data collection', () => {
  beforeEach(() => {
    customLogger = customLogger.mockImplementation(logger);
  });

  afterEach(() => {
    customLogger.mockReset();
  });

  pluginTester({
    ...commonOpts,
    tests: [
      {
        title: 'calls custom logger with data for `targetImports` only',
        code: `import React from 'react';
import { B200, N0, N70A } from '@atlaskit/theme/colors';
export default {
  B200,
  N0,
  N70A
};`,
        pluginOptions: {
          targetImports: '@atlaskit/theme/colors',
          customLogger,
        },
        teardown() {
          expect(customLogger).toBeCalledWith(
            new Map().set('', {
              BaseProjectName: '@atlaskit/fabric',
              DependencyVersions: {},
              File: '',
              ImportDeclarations: {
                '@atlaskit/theme/colors': ['B200', 'N0', 'N70A'],
              },
              Project: '@atlaskit/fabric',
            }),
          );
        },
      } as TestObject,
      {
        title: 'calls custom logger with data for CallExpression type',
        code: `import React from 'react';
import { themeFunction } from '@atlaskit/theme';
export default (props) => themeFunction(props);`,
        pluginOptions: {
          targetImports: '@atlaskit/theme',
          customLogger,
        },
        teardown() {
          expect(customLogger).toBeCalledWith(
            new Map().set('', {
              BaseProjectName: '@atlaskit/fabric',
              DependencyVersions: {},
              File: '',
              ImportDeclarations: { '@atlaskit/theme': ['themeFunction'] },
              CallExpressions: [
                {
                  end: 115,
                  innerComments: null,
                  leadingComments: null,
                  loc: {
                    end: {
                      column: 39,
                      line: 3,
                    },
                    identifierName: 'themeFunction',
                    start: {
                      column: 26,
                      line: 3,
                    },
                  },
                  name: 'themeFunction',
                  start: 102,
                  trailingComments: null,
                  type: 'Identifier',
                },
              ],
              Project: '@atlaskit/fabric',
            }),
          );
        },
      } as TestObject,
      {
        title: 'calls custom logger with data for MemberExpression type',
        code: `import React from 'react';
import { colors } from '@atlaskit/theme';
export default () => colors.B200;`,
        pluginOptions: {
          targetImports: '@atlaskit/theme',
          customLogger,
        },
        teardown() {
          expect(customLogger).toBeCalledWith(
            new Map().set('', {
              BaseProjectName: '@atlaskit/fabric',
              DependencyVersions: {},
              File: '',
              ImportDeclarations: { '@atlaskit/theme': ['colors'] },
              MemberExpressions: [
                {
                  object: {
                    end: 96,
                    innerComments: null,
                    leadingComments: null,
                    loc: {
                      end: {
                        column: 27,
                        line: 3,
                      },
                      identifierName: 'colors',
                      start: {
                        column: 21,
                        line: 3,
                      },
                    },
                    name: 'colors',
                    start: 90,
                    trailingComments: null,
                    type: 'Identifier',
                  },
                  property: {
                    end: 101,
                    innerComments: undefined,
                    leadingComments: undefined,
                    loc: {
                      end: {
                        column: 32,
                        line: 3,
                      },
                      identifierName: 'B200',
                      start: {
                        column: 28,
                        line: 3,
                      },
                    },
                    name: 'B200',
                    start: 97,
                    trailingComments: undefined,
                    type: 'Identifier',
                  },
                },
              ],
              Project: '@atlaskit/fabric',
            }),
          );
        },
      } as TestObject,
      {
        title: 'calls custom logger with data for JSXElement type',
        code: `import React from 'react';
import Avatar from '@atlaskit/avatar';
export default () => <Avatar />;`,
        pluginOptions: {
          targetImports: '@atlaskit/avatar',
          customLogger,
        },
        teardown() {
          expect(customLogger).toBeCalledWith(
            new Map().set('', {
              BaseProjectName: '@atlaskit/fabric',
              DependencyVersions: {},
              File: '',
              ImportDeclarations: { '@atlaskit/avatar': ['Avatar'] },
              JSXElements: [
                {
                  attributes: [],
                  locationEnd: 97,
                  locationStart: 87,
                  nodePath: 'Avatar',
                  reference: 'Avatar',
                },
              ],
              Project: '@atlaskit/fabric',
            }),
          );
        },
      } as TestObject,
      {
        title:
          'calls custom logger with data for JSXElement type with attributes',
        code: `import React from 'react';
import Avatar from '@atlaskit/avatar';
export default () => <Avatar href="/path/to/item.html" />;`,
        pluginOptions: {
          targetImports: '@atlaskit/avatar',
          customLogger,
        },
        teardown() {
          expect(customLogger).toBeCalledWith(
            new Map().set('', {
              BaseProjectName: '@atlaskit/fabric',
              DependencyVersions: {},
              File: '',
              ImportDeclarations: { '@atlaskit/avatar': ['Avatar'] },
              JSXElements: [
                {
                  attributes: ['href="/path/to/item.html"'],
                  locationEnd: 123,
                  locationStart: 87,
                  nodePath: 'Avatar',
                  reference: 'Avatar',
                },
              ],
              Project: '@atlaskit/fabric',
            }),
          );
        },
      } as TestObject,
      {
        title: 'calls custom logger with data for nested JSXElement type',
        code: `import React from 'react';
import Avatar from '@atlaskit/avatar';
export default () => <div><Avatar /></div>;`,
        pluginOptions: {
          targetImports: '@atlaskit/avatar',
          customLogger,
        },
        teardown() {
          expect(customLogger).toBeCalledWith(
            new Map().set('', {
              BaseProjectName: '@atlaskit/fabric',
              DependencyVersions: {},
              File: '',
              ImportDeclarations: { '@atlaskit/avatar': ['Avatar'] },
              JSXElements: [
                {
                  attributes: [],
                  locationEnd: 102,
                  locationStart: 92,
                  nodePath: 'div -> Avatar',
                  reference: 'Avatar',
                },
              ],
              Project: '@atlaskit/fabric',
            }),
          );
        },
      } as TestObject,
    ],
  });
});

describe('Transforms', () => {
  const { babelOptions, ...transformOpts } = commonOpts;

  describe('trackOnClient', () => {
    pluginTester({
      ...transformOpts,
      tests: [
        {
          title:
            'injects data-cognize and data-cognize-tracer in every JSX opening element',
          code: `import React from 'react';
import Avatar from '@atlaskit/avatar';
const Link = () => <a href="/blah.html">Link</a>;
export default () => {
  return <div><Link /><Avatar /></div>;
}`,
          pluginOptions: {
            targetImports: '@atlaskit/avatar',
            customLogger: null,
            trackOnClient: true,
          },
        } as TestObject,
      ],
    });
  });
});

describe('Plugins', () => {
  const { babelOptions, ...transformOpts } = commonOpts;

  let testVar = false;

  let cognizeInstance: BabelPlugin;

  type CustomPlugin = BabelPlugin & {
    testCache: PluginCache;
  };

  const testPlugin = function (this: CustomPlugin) {
    this.testCache = new Map();
    cognizeInstance = this;
    return {
      // Spread visitor to keep all default Cognize visitors
      // ...cognize.visitor,
      // or choose what visitors to keep
      ImportDeclaration: this.visitor?.ImportDeclaration,
      Identifier: {
        enter: [
          function Identifier() {
            testVar = true;
          },
        ],
      },
    };
  };

  describe('calls custom plugin', () => {
    pluginTester({
      ...transformOpts,
      tests: [
        {
          title: 'calls plugin function',
          code: `import React from 'react';
import Avatar from '@atlaskit/avatar';
export default () => {
  return <div><Avatar /></div>;
}`,
          pluginOptions: {
            targetImports: '@atlaskit/avatar',
            customLogger: null,
            customVisitors: testPlugin,
          },
          teardown() {
            expect(testVar).toEqual(true);
            expect(cognizeInstance).toHaveProperty('testCache');
          },
        } as TestObject,
      ],
    });
  });
});
