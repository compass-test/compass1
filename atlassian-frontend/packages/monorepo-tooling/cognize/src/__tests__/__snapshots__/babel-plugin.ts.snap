// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data collection @atlaskit/cognize calls custom logger with data for \`targetImports\` only: calls custom logger with data for \`targetImports\` only 1`] = `

import React from 'react';
import { B200, N0, N70A } from '@atlaskit/theme/colors';
export default {
  B200,
  N0,
  N70A
};

      ↓ ↓ ↓ ↓ ↓ ↓

"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _colors = require("@atlaskit/theme/colors");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  B200: _colors.B200,
  N0: _colors.N0,
  N70A: _colors.N70A
};
exports.default = _default;

`;

exports[`Data collection @atlaskit/cognize calls custom logger with data for CallExpression type: calls custom logger with data for CallExpression type 1`] = `

import React from 'react';
import { themeFunction } from '@atlaskit/theme';
export default (props) => themeFunction(props);

      ↓ ↓ ↓ ↓ ↓ ↓

"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _theme = require("@atlaskit/theme");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = props => (0, _theme.themeFunction)(props);

exports.default = _default;

`;

exports[`Data collection @atlaskit/cognize calls custom logger with data for JSXElement type with attributes: calls custom logger with data for JSXElement type with attributes 1`] = `

import React from 'react';
import Avatar from '@atlaskit/avatar';
export default () => <Avatar href="/path/to/item.html" />;

      ↓ ↓ ↓ ↓ ↓ ↓

"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _avatar = _interopRequireDefault(require("@atlaskit/avatar"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = () => /*#__PURE__*/_react.default.createElement(_avatar.default, {
  href: "/path/to/item.html"
});

exports.default = _default;

`;

exports[`Data collection @atlaskit/cognize calls custom logger with data for JSXElement type: calls custom logger with data for JSXElement type 1`] = `

import React from 'react';
import Avatar from '@atlaskit/avatar';
export default () => <Avatar />;

      ↓ ↓ ↓ ↓ ↓ ↓

"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _avatar = _interopRequireDefault(require("@atlaskit/avatar"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = () => /*#__PURE__*/_react.default.createElement(_avatar.default, null);

exports.default = _default;

`;

exports[`Data collection @atlaskit/cognize calls custom logger with data for MemberExpression type: calls custom logger with data for MemberExpression type 1`] = `

import React from 'react';
import { colors } from '@atlaskit/theme';
export default () => colors.B200;

      ↓ ↓ ↓ ↓ ↓ ↓

"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _theme = require("@atlaskit/theme");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = () => _theme.colors.B200;

exports.default = _default;

`;

exports[`Data collection @atlaskit/cognize calls custom logger with data for nested JSXElement type: calls custom logger with data for nested JSXElement type 1`] = `

import React from 'react';
import Avatar from '@atlaskit/avatar';
export default () => <div><Avatar /></div>;

      ↓ ↓ ↓ ↓ ↓ ↓

"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _avatar = _interopRequireDefault(require("@atlaskit/avatar"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = () => /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement(_avatar.default, null));

exports.default = _default;

`;

exports[`Plugin options @atlaskit/cognize does not call custom logger if not passed through options: does not call custom logger if not passed through options 1`] = `

import React from 'react';
import { B200, N0, N70A } from '@atlaskit/theme/colors';
export default {
  B200,
  N0,
  N70A
};

      ↓ ↓ ↓ ↓ ↓ ↓

"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _colors = require("@atlaskit/theme/colors");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  B200: _colors.B200,
  N0: _colors.N0,
  N70A: _colors.N70A
};
exports.default = _default;

`;

exports[`Plugin options @atlaskit/cognize does not parse file if \`ignoreDirs\` matches filename: does not parse file if \`ignoreDirs\` matches filename 1`] = `

// eslint-disable-next-line import/no-extraneous-dependencies
import { B200, N0, N70A } from '@atlaskit/theme/colors';

export default {
  B200,
  N0,
  N70A,
};

      ↓ ↓ ↓ ↓ ↓ ↓

"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _colors = require("@atlaskit/theme/colors");

// eslint-disable-next-line import/no-extraneous-dependencies
var _default = {
  B200: _colors.B200,
  N0: _colors.N0,
  N70A: _colors.N70A
};
exports.default = _default;

`;

exports[`Plugin options @atlaskit/cognize loads custom visitor: loads custom visitor 1`] = `

import React from 'react';
import { B200, N0, N70A } from '@atlaskit/theme/colors';
const myFunc = () => B200();
myFunc();
export default {
  B200,
  N0,
  N70A
};

      ↓ ↓ ↓ ↓ ↓ ↓

"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _colors = require("@atlaskit/theme/colors");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const myFunc = () => (0, _colors.B200)();

myFunc();
var _default = {
  B200: _colors.B200,
  N0: _colors.N0,
  N70A: _colors.N70A
};
exports.default = _default;

`;

exports[`Plugins calls custom plugin @atlaskit/cognize calls plugin function: calls plugin function 1`] = `

import React from 'react';
import Avatar from '@atlaskit/avatar';
export default () => {
  return <div><Avatar /></div>;
}

      ↓ ↓ ↓ ↓ ↓ ↓

import React from 'react';
import Avatar from '@atlaskit/avatar';
export default (() => {
  return <div><Avatar /></div>;
});

`;

exports[`Transforms trackOnClient @atlaskit/cognize injects data-cognize and data-cognize-tracer in every JSX opening element: injects data-cognize and data-cognize-tracer in every JSX opening element 1`] = `

import React from 'react';
import Avatar from '@atlaskit/avatar';
const Link = () => <a href="/blah.html">Link</a>;
export default () => {
  return <div><Link /><Avatar /></div>;
}

      ↓ ↓ ↓ ↓ ↓ ↓

var _jsxFileName = "";
import React from 'react';
import Avatar from '@atlaskit/avatar';

const Link = () => <a href="/blah.html" data-cognize-tracer={function () {
  window.__COGNIZE__ = window.__COGNIZE__ || {};
  window.__COGNIZE__["_a_3_"] = {
    imports: "@atlaskit/avatar,Avatar",
    fileName: "",
    projectName: "@atlaskit/fabric",
    start: 3,
    column: 19,
    el: () => document.querySelector('[data-cognize=' + "_a_3_")
  };
  return;
}()} data-cognize={"_a_3_"}>Link</a>;

export default (() => {
  return <div data-cognize-tracer={function () {
    window.__COGNIZE__ = window.__COGNIZE__ || {};
    window.__COGNIZE__["_div_5_"] = {
      imports: "@atlaskit/avatar,Avatar",
      fileName: "",
      projectName: "@atlaskit/fabric",
      start: 5,
      column: 9,
      el: () => document.querySelector('[data-cognize=' + "_div_5_")
    };
    return;
  }()} data-cognize={"_div_5_"}><Link data-cognize-tracer={function () {
      window.__COGNIZE__ = window.__COGNIZE__ || {};
      window.__COGNIZE__["_Link_5_"] = {
        imports: "@atlaskit/avatar,Avatar",
        fileName: "",
        projectName: "@atlaskit/fabric",
        start: 5,
        column: 14,
        el: () => document.querySelector('[data-cognize=' + "_Link_5_")
      };
      return;
    }()} data-cognize={"_Link_5_"} /><Avatar data-cognize-tracer={function () {
      window.__COGNIZE__ = window.__COGNIZE__ || {};
      window.__COGNIZE__["_Avatar_5_"] = {
        imports: "@atlaskit/avatar,Avatar",
        fileName: "",
        projectName: "@atlaskit/fabric",
        start: 5,
        column: 22,
        el: () => document.querySelector('[data-cognize=' + "_Avatar_5_")
      };
      return;
    }()} data-cognize={"_Avatar_5_"} /></div>;
});

`;
